test/app/background/reducers/cosmeticFilterReducerTest.ts

```diff
@@ -0,0 +1,474 @@
+  const origin = 'https://brave.com'
+  const state: State = deepFreeze({
+    tabs: {
+      2: {
+        origin,
+        hostname: 'brave.com',
+        adsBlocked: 0,
+        controlsOpen: true,
+        braveShields: 'allow',
+        trackersBlocked: 0,
+        httpsRedirected: 0,
+        javascriptBlocked: 0,
+        fingerprintingBlocked: 0,
+        id: 2,
+        httpUpgradableResources: 'block',
+        javascript: 'block',
+        trackers: 'block',
+        ads: 'block',
+        fingerprinting: 'block',
+        cookies: 'block',
+        noScriptInfo: {},
+        adsBlockedResources: [],
+        fingerprintingBlockedResources: [],
+        httpsRedirectedResources: [],
+        javascriptBlockedResources: [],
+        trackersBlockedResources: []
+      }
+    },
+    windows: {
+      1: 2
+    },
+    currentWindowId: 1
+  })
+  describe('SHIELDS_PANEL_DATA_UPDATED', function () {
+    it('updates state detail', function () {
+      const tabId = 2
+      const details: ShieldDetails = {
+        id: tabId,
+        hostname: 'brave.com',
+        origin: 'brave.com',
+        ads: 'block',
+        trackers: 'block',
+        httpUpgradableResources: 'block',
+        javascript: 'block',
+        fingerprinting: 'block',
+        cookies: 'block'
+      }
+      assert.deepEqual(
+          shieldsPanelReducer(initialState.shieldsPanel, {
+            type: shieldPanelTypes.SHIELDS_PANEL_DATA_UPDATED,
+            details
+          }), {
+            currentWindowId: -1,
+            tabs: {
+              [tabId]: {
+                adsBlocked: 0,
+                trackersBlocked: 0,
+                httpsRedirected: 0,
+                javascriptBlocked: 0,
+                fingerprintingBlocked: 0,
+                hostname: 'brave.com',
+                origin: 'brave.com',
+                id: tabId,
+                ads: 'block',
+                trackers: 'block',
+                httpUpgradableResources: 'block',
+                javascript: 'block',
+                fingerprinting: 'block',
+                cookies: 'block',
+                controlsOpen: true,
+                braveShields: 'allow',
+                noScriptInfo: {},
+                adsBlockedResources: [],
+                fingerprintingBlockedResources: [],
+                httpsRedirectedResources: [],
+                javascriptBlockedResources: [],
+                trackersBlockedResources: []
+              }
+            },
+            windows: {}
+          })
+    })
+  })
+  describe('SHIELDS_TOGGLED', function () {
+    before(function () {
+      this.reloadTabSpy = sinon.spy(tabsAPI, 'reloadTab')
+      this.setAllowBraveShieldsSpy = sinon.spy(shieldsAPI, 'setAllowBraveShields')
+    })
+    after(function () {
+      this.reloadTabSpy.restore()
+      this.setAllowBraveShieldsSpy.restore()
+    })
+    it('should call setAllowBraveShields', function () {
+      assert.deepEqual(
+            shieldsPanelReducer(state, {
```

Comment:
nit: not sure why tslint didn't complain but this extra spacing doesn't look good. having it indented like https://github.com/brave/brave-extension/pull/48/files#diff-bf9c7b9f5cac8e57d36f4d6559925015R451 seems more appropriate